### ISSUE TITLE
Getting "Limited ads" (Reward Ad)

### ISSUE THREAD

**Issue by mmgzentertainment** [googleads/googleads-mobile-unity#2580]
### [REQUIRED] Step 1: Describe your environment

  * Unity version: 2022.1.22f1
  * Google Mobile Ads Unity plugin version: v7.4.1
  * Platform: Android
  * Platform OS version: Android 12

### [REQUIRED] Step 2: Describe the problem

Getting a lot of "Limited ads" when I use latest plugin version (v7.4.1) even with GDPR UMP consent.
It was fine with Unity Plugin v7.2.0. (no such thing as Limited Ads until v7.2.0) 
(so, basically unity plugin for google Mobile Ads Android SDK 21.0.0 was okay)

#### Steps to reproduce:

May be  gad_has_consent_for_cookies is off in some devices. But it should be enabled by default. developer can be turned on/off by java or objective-c. but there is no way to do that in current latest unity plugin.
If I use v7.2.0, "Limited ads" just goes away for same device ( same phone settings, almost same code flow except for deprecated functions).

#### Relevant Code:

No relevant code. any reward ad code. (Include GDRP UMP consent form)

**Comment by mmgzentertainment** [comment]
I follow the documents. those are empty functions and does nothing? what am I missing here?

**Comment by NVentimiglia** [comment]
Hi @mmgzentertainment 

Please use player prefs.

```
PlayerPrefs.SetInt("gad_has_consent_for_cookies", 0);
PlayerPrefs.Save();
```

**Comment by NVentimiglia** [comment]
Hi @mmgzentertainment 

Update on this issue, we have confirmed this is a problem and are working on extending the API to allow for this functionality.

**Comment by mmgzentertainment** [comment]
Is this fixed in v8.0.0?

**Comment by NVentimiglia** [comment]
@mmgzentertainment 

This is not released in v8.0.0.

**Comment by ericleich** [comment]
@mmgzentertainment Please take a look at https://support.google.com/admob/answer/10105530. It sounds like you upgraded to Unity plugin 7.4.1 and implemented the UMP features to load and show a consent form, effectively using the IAB TCF v2.0 consent framework. From the help center page:

```
If a publisher is using the IAB TCF v2.0 consent framework, we will attempt to serve an eligible limited ad
when there is no consent for Purpose 1 but Google is allowed as a vendor to operate based on legitimate
interest or consent for purposes 2, 7, 9, and 10.
```

If you rollback to 7.2.0, and particularly if the re-install mechanism clears the local storage where the TCF v2 string is written, then you are effectively not using the IAB TCF v2.0 consent framework anymore and the above would not apply.

If I'm reading this correctly, this is working as intended.
